lib/collection/src/shards/update_tracker.rs

```diff
@@ -0,0 +1,35 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+
+#[derive(Clone, Debug, Default)]
+pub struct UpdateTracker {
+    update_operations: Arc<AtomicUsize>,
+}
+
+impl UpdateTracker {
+    pub fn is_update_in_progress(&self) -> bool {
```

Comment:
What about adding a comment here explaining what this is intended for?

```suggestion
    /// True if an update is currently being applied to a shard
    /// 
    /// During this time, the shard is write-locked preventing any search operations.
    /// Then, it may be faster to fan out to a different shard replica instead.
    pub fn is_update_in_progress(&self) -> bool {
```